src/github.com/couchbase/sync_gateway/db/crud.go

```diff
@@ -598,7 +602,9 @@ func (db *Database) updateDoc(docid string, allowImport bool, callback func(*doc
 				if curBody, err = db.getAvailableRev(doc, doc.CurrentRev); curBody != nil {
 					base.LogTo("CRUD+", "updateDoc(%q): Rev %q causes %q to become current again",
 						docid, newRevID, doc.CurrentRev)
-					channels, access, roles, err = db.getChannelsAndAccess(doc, curBody, doc.CurrentRev)
+					channelSet, access, roles, oldBodyBytes, err = db.getChannelsAndAccess(doc, curBody, doc.CurrentRev)
+
+					oldBodyJSON = string(oldBodyBytes)
```

Comment:
On line 524 the use of short variable declaration causes a locally scoped instance of oldBodyBytes to be created, which is also written to on line 605.

To set the value on the instance scoped at the outer level of the method these are the options I think we have:

Define channelSet, access, roles, oldBodyBytes, err as var's at the top of the method and use simple assignment on line 524 rather than short variable declaration

Return a []byte from getChannelsAndAccess and set oldBodyJSON string value with a type conversion

Return a string from getChannelsAndAccess and set oldBodyJSON string value with an assignment of returned value